security/pkg/pki/ra/k8s_ra.go

```diff
@@ -34,11 +34,24 @@ import (
 
 // KubernetesRA integrated with an external CA using Kubernetes CSR API
 type KubernetesRA struct {
-	csrInterface                 clientset.Interface
-	keyCertBundle                *util.KeyCertBundle
-	raOpts                       *IstioRAOptions
+	csrInterface clientset.Interface
+
+	// has roots from ./etc/external-ca-cert/root-cert.pem (if mounted).
+	// May fallback to legacy-unknown or have no roots.
+	// This does not have any key or cert - only roots.
+	keyCertBundle *util.KeyCertBundle
```

Comment:
Is "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt" valid for legacy signer as noted above in `security/pkg/pki/ra/common.go`? We could also clarify this doesn't contain any leaf certs or keys, but may include intermediate certificates in addition to roots (if that is the case).

Reply:
This is the signer used for API server DNS cert - should not be used for anything else ( and we removed the code for the legacy signer ). 

And it only has roots, no intermediates.